compiler/rustc_resolve/src/late.rs

```diff
@@ -1708,7 +1801,15 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                 GenericParamKind::Type { .. } => (&mut function_type_rib, DefKind::TyParam),
                 GenericParamKind::Const { .. } => (&mut function_value_rib, DefKind::ConstParam),
                 GenericParamKind::Lifetime => {
-                    function_lifetime_rib.bindings.insert(ident, ());
+                    let LifetimeRibKind::Generics { parent, .. } = lifetime_kind else { panic!() };
+                    let res = LifetimeRes::Param { param: def_id, binder: parent };
+                    if let Some(prev_region) = self.r.lifetimes_res_map.insert(param.id, res) {
+                        panic!(
+                            "lifetime resolved multiple times ({:?} before, {:?} now)",
+                            prev_region, def_id
+                        )
```

Comment:
I suspect that this check is relevant to all places where we attempt to insert something to `lifetimes_res_map`.
So maybe it's better to create an analogue of `fn record_partial_res` but for lifetimes instead of inserting such panics randomly.